### ISSUE TITLE
`replaces_id` is not working with `naughty.notification{}` but working with `naughty.notify{}`

### ISSUE THREAD

**Issue by actionless** [awesomeWM/awesome#2822]
Output of `awesome --version`:

```
awesome v4.3-291-gb7956718-dirty (Too long)
 • Compiled against Lua 5.1.4 (running with LuaJIT 2.0.5)
 • D-Bus support: ✔
 • xcb-errors support: ✔
 • execinfo support: ✔
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
```

**How to reproduce the issue:**

This works:

```lua
#!/bin/sh
_=[[
exec awesome-client "dofile('$(readlink -e $0)')"
]]

local naughty = require('naughty')

local n1 = naughty.notification{
    title = "1111111",
    text = "111111",
}

local timer
local function second_notification()
  naughty.notify{
  --naughty.notification{
      replaces_id = n1.id,
      title = "222222222",
      text = "22222",
  }
  timer:stop()
end

local gears_timer = require("gears.timer")
timer = gears_timer({
  callback=second_notification,
  timeout=1,
  autostart=true,
  call_now=false,
})

```

And this doesn't work:

```lua
#!/bin/sh
_=[[
exec awesome-client "dofile('$(readlink -e $0)')"
]]

local naughty = require('naughty')

local n1 = naughty.notification{
    title = "1111111",
    text = "111111",
}

local timer
local function second_notification()
  --naughty.notify{
  naughty.notification{
      replaces_id = n1.id,
      title = "222222222",
      text = "22222",
  }
  timer:stop()
end

local gears_timer = require("gears.timer")
timer = gears_timer({
  callback=second_notification,
  timeout=1,
  autostart=true,
  call_now=false,
})

```

**Comment by vn-ki** [comment]
@actionless Refer https://github.com/awesomeWM/awesome/pull/2816#issuecomment-509778842

> This is normal, the new API doesn't have a replace_id. You are creating a new notification object, which is a new notification. You need to update the old notification object (my_old_notification.message = "foo"). naughty.notify is a proxy that may or may not create new objects, naughty.notification() is directly the constructor of a new notification.

**Comment by vn-ki** [comment]
@psychon This can be closed now as #2823 undocumented `replaces_id`
